pkg/util/testing/core.go

```diff
@@ -47,26 +46,23 @@ func SingleContainerForRequest(request map[corev1.ResourceName]string) []corev1.
 	}
 }
 
-// CheckLatestEvent will return true if the latest event is as you want.
-func CheckLatestEvent(ctx context.Context, k8sClient client.Client,
-	eventReason string,
-	eventType string, eventNote string) (bool, error) {
-	events := &eventsv1.EventList{}
-	if err := k8sClient.List(ctx, events, &client.ListOptions{}); err != nil {
+// CheckEventRecordedFor checks if an event identified by eventReason, eventType, eventNote
+// was recorded for the object indentified by ref.
+func CheckEventRecordedFor(ctx context.Context, k8sClient client.Client,
+	eventReason string, eventType string, eventMessage string,
+	ref types.NamespacedName) (bool, error) {
+	events := &corev1.EventList{}
+	if err := k8sClient.List(ctx, events, client.InNamespace(ref.Namespace)); err != nil {
 		return false, err
 	}
 
-	length := len(events.Items)
-	if length == 0 {
-		return false, errors.New("no events currently exist")
-	}
-
-	item := events.Items[length-1]
-	if item.Reason == eventReason && item.Type == eventType && item.Note == eventNote {
-		return true, nil
+	for i := range events.Items {
+		item := &events.Items[i]
+		if item.InvolvedObject.Name == ref.Name && item.Reason == eventReason && item.Type == eventType && item.Message == eventMessage {
+			return true, nil
+		}
 	}
-
-	return false, fmt.Errorf("mismatch with the latest event: got r:%v t:%v n:%v, reg %v", item.Reason, item.Type, item.Note, item.Regarding)
+	return false, fmt.Errorf("event not found fater checking %d events", len(events.Items))
```

Comment:
Include the `eventReason`, `eventType`, `eventMessage` and namespace in the error message.

Reply:
done